pyo3-ffi/src/cpython/descrobject.rs

```diff
@@ -53,7 +53,7 @@ pub struct PyMethodDescrObject {
 #[repr(C)]
 pub struct PyMemberDescrObject {
     pub d_common: PyDescrObject,
-    pub d_member: *mut PyGetSetDef,
+    pub d_member: *mut PyMemberDef,
```

Comment:
It looks like this changed in https://github.com/python/cpython/commit/042f31da552c19054acd3ef7bb6cfd857bce172b, which was first released as part of Python 3.11 (I think, please double-check that).

In which case, I think the correct fix is to make this conditional on Python version.